### ISSUE TITLE
Retest: Unselected configurations available for evaluation

### ISSUE THREAD

**Issue by ibarra-michelle** [US-EPA-CAMD/PI-1#6434]
After selecting one configuration to export and import for evaluation, the evaluation module displays a second or duplicate configuration from the same facility (ORIS), even only one configuration was selected for evaluation. 

This bug was observed in the ORIS Config combinations listed below. The timestamp is also between the selected and unselected (or duplicate) configuration.

1.ORIS 2514 Config 40 was selected (some how, config 50 data was also available to be evaluated)  
![Image](https://github.com/user-attachments/assets/8c28cf0e-c955-48cd-8d50-0b1f15021ad8)

2. ORIS 628 Config 1 was checked out (some how, config 2 data was also available to be evaluated)  
![Image](https://github.com/user-attachments/assets/5dac7de5-bc9a-49b3-b68e-112b680a6c3f)

3. ORIS 50282 Config PR003, PR004, PR005, CSPR06 was checked out (some how, duplicate records of PR003, PR004, PR005, CSPR06 were made available to evaluate. I reverted to the official record and one of the two MP records were removed. The screenshot displays the remaining record with no configuration associated with it.). 
![Image](https://github.com/user-attachments/assets/9a72b4fc-0ede-40ef-b69a-9546566ccc62)

**Comment by annalbrecht** [comment]
Blocked by #6425

**Comment by maxdiebold-erg** [comment]
@ibarra-michelle I looked into the issue in the first point, and it appears that both of those units (40 & 50) have been retired, and the associated Monitor Plans _should_ have ended. As I noted previously, the new import process needs to assess all current plans in the workspace, and it determined that those two plans should have ended. I'd guess the same thing is happening in the point number 2, but I will surely double check.

We probably want to add logic to end a USC and monitor plan when a unit is retired.

**EDIT:** I've confirmed that units 1 & 2 in ORIS 628 have also been retired.

**Comment by ibarra-michelle** [comment]
> We probably want to add logic to end a USC and monitor plan when a unit is retired.

Thanks @maxdiebold-erg for summarizing your findings and including the recommended next step. Let's confirm what are the next steps are at the next Standup meeting (e.g., who is writing up the new ticket, initial requirements, do we close this tix #6434, priority, etc.).

FYA - @annalbrecht , @spetros-do , @alangmaid , @maheese , @bryanramirez1 , @j-tafoya

**Comment by ibarra-michelle** [comment]
> We probably want to add logic to end a USC and monitor plan when a unit is retired.
> 

From the 10/30/2024 DSU meeting, the team concluded to create 2 new placeholder tickets (requirements needed for both tickets) to address Max's recommendation to to add logic to end a USC and monitor plan when a unit is retired:

1. The 1st tix (#6437) should address the short-term solution (script) in 2025Q1 that Must be included in the ECMPS.
2. The 2nd tix (#6438) should address a long-term solution Post ECMPS release (e.g., possibly development of functionality in the Admin view or DS/DP Workspace view).

**Comment by ibarra-michelle** [comment]
FYA @esaber76 @maxdiebold-erg 
Summary: Unable to verify in the test environment. For the [known retired configurations](https://github.com/US-EPA-CAMD/easey-ui/issues/6434#issuecomment-2444971797) (1. ORIS 2517 Config 40 and Config 50 and 2. ORIS 628 Config 1 and 2), the "Revert to Official Record" changes the status from "Inactive" to "Active"

## ORIS 2517 Config 40 and Config 50 
1. I applied the "Revert to official record"  for both ORIS 2517 Config 40 and Config 50 so they were no longer available in the evaluation queue. I checked out ORIS 2517 Config 40, exported/imported the MP json file, navigated back to the Evaluation module, and see ORIS 2517 Config 40 and Config 50 are available for evaluation again (screenshot 1).

### Screenshot 1 - Config 40 and 50 available for eval
<img width="668" alt="Image" src="https://github.com/user-attachments/assets/d2ed6ef9-c058-40ab-a695-ed3892e1cf7f" />

## ORIS 628 Config 1 and 2
2. ORIS 628 Config 1 and 2 - Before revert back to official, both configs have the the Status listed as "Inactive" (screenshot 2). After I revert, both of the configs status change to "Active" (screenshot 2). ORIS 628 Config 1, exported/imported the MP json file, navigated back to the Evaluation module, and see ORIS 628 Config 1 and 2 are available for evaluation again (screenshot 4).

### Screenshot 2 - Inactive status
<img width="278" alt="Image" src="https://github.com/user-attachments/assets/f6c89e27-00a0-4c00-aa9b-5e652b4f09ed" />

### Screenshot 3 - Active status
<img width="424" alt="Image" src="https://github.com/user-attachments/assets/8585076e-b402-4066-8942-2c87db59d29c" />

### Screenshot 4 - Eval queue
<img width="666" alt="Image" src="https://github.com/user-attachments/assets/890b6bd8-cb8f-4a90-9658-baad874581ff" />

## ORIS 50282 Config PR003, PR004, PR005, CSPR06
3. ORIS 50282 Config PR003, PR004, PR005, CSPR06, the configuration with no values still remains available (Screenshot 4). After reverting ORIS 50282 Config PR003, PR004, PR005, CSPR06, exporting/importing the MP json file, navigating back to the Evaluation module, I see both ORIS 50282 Config PR003, PR004, PR005, CSPR06 and the Configuration with no value available to evaluate (screenshot 5).

### Screenshot 4 - Config with no values
<img width="449" alt="Image" src="https://github.com/user-attachments/assets/d42c7b63-9c22-4cc9-a22a-6531bfe4cdb4" />

### Screenshot 5 - Eval queue
<img width="628" alt="Image" src="https://github.com/user-attachments/assets/f3401cfc-bc60-455b-9f53-02ff4cd583a1" />

**Comment by maxdiebold-erg** [comment]
Thanks @ibarra-michelle, I will look into your third point. 

The change from "Active" to "Inactive" is due to the plan ending, as explained [above](https://github.com/US-EPA-CAMD/easey-ui/issues/6434#issuecomment-2444971797).

**Comment by esaber76** [comment]
@ibarra-michelle @maxdiebold-erg - we probably should have put these MP's back to their original states in the database before retesting. I can do that if needed, but probably not until next week.

**Comment by ibarra-michelle** [comment]
> [@ibarra-michelle](https://github.com/ibarra-michelle) [@maxdiebold-erg](https://github.com/maxdiebold-erg) - we probably should have put these MP's back to their original states in the database before retesting. I can do that if needed, but probably not until next week.

@esaber76 @maxdiebold-erg maybe we can check-in at Monday's stand up on whether to pursue Scott's database option?

**Comment by esaber76** [comment]
> > [@ibarra-michelle](https://github.com/ibarra-michelle) [@maxdiebold-erg](https://github.com/maxdiebold-erg) - we probably should have put these MP's back to their original states in the database before retesting. I can do that if needed, but probably not until next week.
> 
> [@esaber76](https://github.com/esaber76) [@maxdiebold-erg](https://github.com/maxdiebold-erg) maybe we can check-in at Monday's stand up on whether to pursue Scott's database option?

I would certainly lean towards that. The data was likely already out of sync and in a bad state before retesting occurred. I had been running a series of delete/update statements on prior problems and configurations I knew I was going to retest before the ticket came back. I didn't do anything with these...

**Comment by ibarra-michelle** [comment]
> > > [@ibarra-michelle](https://github.com/ibarra-michelle) [@maxdiebold-erg](https://github.com/maxdiebold-erg) - we probably should have put these MP's back to their original states in the database before retesting. I can do that if needed, but probably not until next week.
> > 
> > 
> > [@esaber76](https://github.com/esaber76) [@maxdiebold-erg](https://github.com/maxdiebold-erg) maybe we can check-in at Monday's stand up on whether to pursue Scott's database option?
> 
> I would certainly lean towards that. The data was likely already out of sync and in a bad state before retesting occurred. I had been running a series of delete/update statements on prior problems and configurations I knew I was going to retest before the ticket came back. I didn't do anything with these...

@esaber76 Great, we'll check-in on Tuesday's ECMPS Priorities meeting as we have the holiday on Monday. Thank you!

**Comment by esaber76** [comment]
@ibarra-michelle - this is ready to retest. 

1. ORIS 2517, 40 and 50. #6437 will fix this pre-production release. These units retired Q3 2012 and the Configuration Management changes are trying to end them (which is appropriate and would require evaluations 1st). I did what #6437 will do and added end_rpt_period_id's to the affected tables in official, reverted, and they were set to inactive (as should be). The real test now is to export/import an active MP for this facility and ensure these retired MP's do not need to be evaluated.
2. ORIS 628, 1 and 2. Your revert set them back to active (this is what was in official). All in all, similar situation to above. These units retired in Q4 2018. I again did what #6437 will do and ended them in official and reverted. They should be inactive now. The test now is to export/import an active MP for this facility and ensure these retired MP's do not need to be evaluated.
3. ORIS 50282, CSPR06, PR003, PR004, PR005. The garbage MP created back in October still existed in the workspace tables. I got rid of it. You can proceed with normal testing for this configuration.

I stole the plants in CBS. You will need to take them back.

**Comment by ibarra-michelle** [comment]
Verified in the test env that all configurations did not make unselected configurations available for evaluation:

1. ORIS 2514, 40 and 50
2. ORIS 628, 1 and 2
3. ORIS 50282, CSPR06, PR003, PR004, PR005
